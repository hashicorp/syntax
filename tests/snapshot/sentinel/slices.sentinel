# https://developer.hashicorp.com/sentinel/docs/language/slices
a = [1, 2, 3, 4, 5]
b = a[1:4]

a = "hello"
b = a[1:4]

b = a[:2]
b = a[2:]
