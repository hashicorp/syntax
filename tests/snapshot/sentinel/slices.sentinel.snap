># https://developer.hashicorp.com/sentinel/docs/language/slices
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.sentinel comment.line.number-sign.sentinel
>a = [1, 2, 3, 4, 5]
#^^ source.sentinel
#  ^ source.sentinel keyword.operator.symbol.sentinel
#   ^^ source.sentinel
#     ^ source.sentinel constant.numeric.number.sentinel
#      ^^ source.sentinel
#        ^ source.sentinel constant.numeric.number.sentinel
#         ^^ source.sentinel
#           ^ source.sentinel constant.numeric.number.sentinel
#            ^^ source.sentinel
#              ^ source.sentinel constant.numeric.number.sentinel
#               ^^ source.sentinel
#                 ^ source.sentinel constant.numeric.number.sentinel
#                  ^^ source.sentinel
>b = a[1:4]
#^^ source.sentinel
#  ^ source.sentinel keyword.operator.symbol.sentinel
#   ^^^ source.sentinel
#      ^ source.sentinel constant.numeric.number.sentinel
#       ^ source.sentinel
#        ^ source.sentinel constant.numeric.number.sentinel
#         ^^ source.sentinel
>
>a = "hello"
#^^ source.sentinel
#  ^ source.sentinel keyword.operator.symbol.sentinel
#   ^ source.sentinel
#    ^ source.sentinel string.quoted.double.untitled
#     ^^^^^ source.sentinel string.quoted.double.untitled
#          ^ source.sentinel string.quoted.double.untitled
>b = a[1:4]
#^^ source.sentinel
#  ^ source.sentinel keyword.operator.symbol.sentinel
#   ^^^ source.sentinel
#      ^ source.sentinel constant.numeric.number.sentinel
#       ^ source.sentinel
#        ^ source.sentinel constant.numeric.number.sentinel
#         ^^ source.sentinel
>
>b = a[:2]
#^^ source.sentinel
#  ^ source.sentinel keyword.operator.symbol.sentinel
#   ^^^^ source.sentinel
#       ^ source.sentinel constant.numeric.number.sentinel
#        ^^ source.sentinel
>b = a[2:]
#^^ source.sentinel
#  ^ source.sentinel keyword.operator.symbol.sentinel
#   ^^^ source.sentinel
#      ^ source.sentinel constant.numeric.number.sentinel
#       ^^^ source.sentinel
>